components/omega/test/ocn/StateTest.cpp

```diff
@@ -164,181 +260,216 @@ int main(int argc, char *argv[]) {
+      // Call initialization routine to create default state and other
+      // quantities
       int Err = initStateTest();
       if (Err != 0)
          LOG_CRITICAL("State: Error initializing");
 
-      // Get MPI vars if needed
-      OMEGA::MachEnv *DefEnv = OMEGA::MachEnv::getDefault();
-      MPI_Comm Comm          = DefEnv->getComm();
-      OMEGA::I4 MyTask       = DefEnv->getMyTask();
-      OMEGA::I4 NumTasks     = DefEnv->getNumTasks();
-      bool IsMaster          = DefEnv->isMasterTask();
-
-      OMEGA::HorzMesh *DefHorzMesh = OMEGA::HorzMesh::getDefault();
-      OMEGA::Decomp *DefDecomp     = OMEGA::Decomp::getDefault();
-      OMEGA::Halo *DefHalo         = OMEGA::Halo::getDefault();
-
-      // These hard-wired variables need to be upated
-      // with retrivals/config options
-      int NVertLevels = 60;
-      int NTimeLevels = 2;
-
-      // Create dimensions (Horz dims computed in Mesh init)
-      auto VertDim = OMEGA::Dimension::create("NVertLevels", NVertLevels);
-
-      for (int NTimeLevels = 2; NTimeLevels < 5; NTimeLevels++) {
-
-         int NewLevel = 1;
-         int CurLevel = 0;
-
-         // Create "default" state
-         if (NTimeLevels == 2) {
-
-            OMEGA::OceanState::init();
-            OMEGA::OceanState *DefOceanState = OMEGA::OceanState::getDefault();
-
-         } else {
-            OMEGA::OceanState *DefState = OMEGA::OceanState::create(
-                "Default", DefHorzMesh, DefHalo, NVertLevels, NTimeLevels);
-            DefState->loadStateFromFile(DefHorzMesh->MeshFileName, DefDecomp);
-         }
+      // Get default mesh, halo and time data
+      HorzMesh *DefHorzMesh   = HorzMesh::getDefault();
+      Halo *DefHalo           = Halo::getDefault();
+      TimeStepper *DefStepper = TimeStepper::getDefault();
+      Clock *ModelClock       = DefStepper->getClock();
+      int NTimeLevelsDef      = DefStepper->getNTimeLevels();
+
+      // Test retrieval of the default state
+      OceanState *DefState = OceanState::getDefault();
+      int NVertLevels      = DefState->NVertLevels;
+      int NCellsAll        = DefState->NCellsAll;
+      int NEdgesAll        = DefState->NEdgesAll;
+      int CurTime          = 0;
+      int NewTime          = 1;
+      if (DefState and NVertLevels > 0) {
+         LOG_INFO("State: Default state retrieval PASS");
+      } else {
+         RetVal += 1;
+         LOG_INFO("State: Default state retrieval FAIL");
+      }
 
-         // Test retrieval of the default state
-         OMEGA::OceanState *DefState = OMEGA::OceanState::get("Default");
-         if (DefState) { // true if non-null ptr
-            LOG_INFO("State: Default state retrieval (NTimeLevels={}) PASS",
-                     NTimeLevels);
-         } else {
-            RetVal += 1;
-            LOG_INFO("State: Default state retrieval (NTimeLevels={}) FAIL",
-                     NTimeLevels);
+      // Get default state arrays and check for reasonable values
```

Comment:
Some of the clean-up in `StateTest.cpp` seems like it would fit better in a separate PR since it isn't related to IOStream integration, but leaving it is fine.

Reply:
It was more than just a cleanup - I had to reorganize so that the default state was left alone. The earlier version kept re-defining the default for the various cases being tested.

Reply:
Sounds good. Did this redefinition cause an issue for IOStreams?

Reply:
It was due to the switch from your loadFromFile to the init stream - since the init stream is only read once on start up we couldn't do repeated reads to initialize state for each of the time step options in the original. So I switched to defining new States and initializing from copies instead.

Reply:
Great, thanks for the explanation!